### DISCUSSION TITLE
Why `Unsafe` methods to make safer unsafe operations?

### DISCUSSION THREAD

**Discussion by scarletquasar** [dotnet/runtime#77694] [General] [answered]
Isn't like a naming mistake?

**Comment by stephentoub** [comment]
Can you clarify the question? `Unsafe` doesn't make things safer; it primarily exposes functionality you can achieve in IL but not in C#, most of which can be used to violate memory/type safety and thus is inherently unsafe.

**Reply by scarletquasar** [reply]
The next reply answered me! Thanks

**Comment by teo-tsirpanis** [comment, marked as answer]
The absence of the `unsafe` keyword does not imply that the code is safe. You might be interested in #31354.
